kernel/src/context.rs

```diff
@@ -226,6 +226,63 @@ impl ReadCString for VmReader<'_, Fallible> {
     }
 }
 
+impl ReadCString for VmReader<'_, Infallible> {
```

Comment:
Is the code (almost) identical with `impl ReadCString for VmReader<'_, Fallible>`? If so, then we need to find a way to avoid the code duplication. One strategy is to convert `VmReader<'_, Infallible>` to `VmReader<'_, Fallible>`, then use the later one's `ReadCString` implementation.

Reply:
Yes, I just copy the `ReadCString` implement here, I will give a try.